machado/management/commands/load_similarity_matches.py

```diff
@@ -51,23 +69,28 @@ def handle(self,
         try:
             feature_file = FeatureLoader(
                 filename=filename,
-                source='InterproScan_source',
+                source=source,
                 organism=organism,
             )
         except ImportingError as e:
             raise CommandError(e)
 
         try:
-            records = SearchIO.parse(file, 'interproscan-xml')
+            records = SearchIO.parse(file, source_type)
         except ValueError as e:
             return CommandError(e)
 
         pool = ThreadPoolExecutor(max_workers=cpu)
         tasks = list()
+        # target is the database name (e.g. nr.dmnd)
+        target = None
         for record in records:
+            # get target from SearchIO's QueryResult if parsing BLAST output
+            if source_type == 'blast-xml':
+                target = record.target
```

Comment:
I think it would be better if you pass source_type to the functionsotre_bio_search_io_hit rather than target.

You just inform the source_type and the function deals with the data.